src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion_img2img.py

```diff
@@ -695,15 +695,19 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
-        # 9. Post-processing
-        image = self.decode_latents(latents)
+        if output_type == "latent":
+            image = latents
+            has_nsfw_concept = None
+        else:
+            # 9. Post-processing
+            image = self.decode_latents(latents)
 
-        # 10. Run safety checker
-        image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
+            # 10. Run safety checker
+            image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
 
-        # 11. Convert to PIL
-        if output_type == "pil":
-            image = self.numpy_to_pil(image)
+            # 11. Convert to PIL
+            if output_type == "pil":
+                image = self.numpy_to_pil(image)
```

Comment:
Can we do this like we do in the alt diffusion pipeline?

https://github.com/huggingface/diffusers/blob/4d0e897c8365bd572cb6da5f8d4033df9e6df6d2/src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py#L661-L678